pkg/util/util.go

```diff
@@ -18,23 +18,31 @@ package util
 
 import (
 	"context"
+	"crypto/rand"
+	"encoding/hex"
+	"fmt"
+	"regexp"
 
 	controlplanev1 "github.com/rancher-sandbox/cluster-api-provider-rke2/controlplane/api/v1alpha1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	clusterv1 "sigs.k8s.io/cluster-api/api/v1beta1"
 	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/log"
 )
 
 func GetOwnerControlPlane(ctx context.Context, c client.Client, obj metav1.ObjectMeta) (*controlplanev1.RKE2ControlPlane, error) {
+	logger := log.FromContext(ctx)
 
 	ownerRefs := obj.OwnerReferences
 	var cpOwnerRef metav1.OwnerReference
 	for _, ownerRef := range ownerRefs {
-		if ownerRef.APIVersion == controlplanev1.GroupVersion.Group && ownerRef.Kind == "RKE2ControlPlane" {
+		logger.V(5).Info("Inside GetOwnerControlPlane", "ownerRef.APIVersion", ownerRef.APIVersion, "ownerRef.Kind", ownerRef.Kind, "cpv1.GroupVersion.Group", controlplanev1.GroupVersion.Group)
+		if ownerRef.APIVersion == controlplanev1.GroupVersion.Group+"/"+controlplanev1.GroupVersion.Version && ownerRef.Kind == "RKE2ControlPlane" {
 			cpOwnerRef = ownerRef
 		}
 	}
 
+	logger.Info("GetOwnerControlPlane result:", "cpOwnerRef", cpOwnerRef)
```

Comment:
Needs a higher verbosity level as it looks like debugging information

Reply:
Would you suggest a level I should use for debug?